engine/src/main/java/org/hibernate/validator/messageinterpolation/ExpressionLanguageFeatureLevel.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.messageinterpolation;
+
+import org.hibernate.validator.Incubating;
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;
+
+/**
+ * Indicates the level of features enabled for the Expression Language engine.
+ *
+ * @since 6.2
+ */
+@Incubating
+public enum ExpressionLanguageFeatureLevel {
+
+	/**
+	 * The default Expression Language feature level.
+	 * <p>
+	 * Depends on the context.
+	 * <p>
+	 * For standard constraint messages, it is {@link ExpressionLanguageFeatureLevel#BEAN_PROPERTIES}.
+	 * <p>
+	 * For custom violations, the default is {@link ExpressionLanguageFeatureLevel#NONE} and, if Expression Language is
+	 * enabled for a given custom violation via the API, the default becomes
+	 * {@link ExpressionLanguageFeatureLevel#VARIABLES} in the context of this given custom violation.
```

Comment:
It's not quite the full story, though.

From what I can see:

* If not enabled explicitly, the default is whatever the user configured through properties, falling back to `NONE`.
* If enabled explicitly, the default is `VARIABLES`, and whatever the user configured through properties is ignored.

Which seems... complex?